pkg/ibmvpc/parser.go

```diff
@@ -529,26 +536,51 @@ func getFipConfig(
 		default:
 			return fmt.Errorf("unsupported fip target : %s", target)
 		}
-		if targetAddress != "" {
-			srcNodes := getCertainNodes(res.Nodes, func(n vpcmodel.Node) bool { return n.Cidr() == targetAddress })
-			routerFip := &FloatingIP{
-				VPCResource: vpcmodel.VPCResource{
-					ResourceName: *fip.Name,
-					ResourceUID:  *fip.CRN,
-					Zone:         *fip.Zone.Name,
-					ResourceType: ResourceTypeFloatingIP},
-				cidr: *fip.Address, src: srcNodes}
-			res.RoutingResources = append(res.RoutingResources, routerFip)
-			res.NameToResource[routerFip.Name()] = routerFip
-
-			// node with fip should not have pgw
-			for _, r := range res.RoutingResources {
-				if pgw, ok := r.(*PublicGateway); ok {
-					// a node captured by a fip should not be captured by a pgw
-					for _, nodeWithFip := range srcNodes {
-						if vpcmodel.HasNode(pgw.Src(), nodeWithFip) {
-							pgw.src = getCertainNodes(pgw.Src(), func(n vpcmodel.Node) bool { return n.Cidr() != nodeWithFip.Cidr() })
-						}
+
+		if targetAddress == "" {
+			continue
+		}
+
+		var srcNodes []vpcmodel.Node
+		var vpcUID string
+		for uid, vpcConfig := range res {
+			srcNodes = getCertainNodes(vpcConfig.Nodes, func(n vpcmodel.Node) bool { return n.Cidr() == targetAddress })
```

Comment:
Is this good enough in a multi-VPC scenario? Are we relying on address spaces to be disjoint?
Wouldn't it be better to compare the target's CRN?
If we do rely on disjoint address spaces, please add here a comment that this is the assumption.

Reply:
Changed to be based on unique id